modules/database/test/std/rec/aiTest.c

```diff
@@ -0,0 +1,429 @@
+    double value = 543.123;
+
+    /* set soft channel */
+    testdbPutFieldOk("test_ai_rec.DTYP", DBF_STRING, "Soft Channel");
+    testdbPutFieldOk("test_ai_rec.INP", DBF_STRING, "test_ai_link_rec.VAL");
+    testdbPutFieldOk("test_ai_link_rec.FLNK", DBF_STRING, "test_ai_rec");
+
+    /* set VAL to on linked record */
+    testdbPutFieldOk("test_ai_link_rec.VAL", DBF_DOUBLE, value);
+
+    /* verify that this record VAL is updated but RVAL is not */
+    testdbGetFieldEqual("test_ai_rec.VAL", DBF_DOUBLE, value);
+    testdbGetFieldEqual("test_ai_rec.RVAL", DBF_DOUBLE, 0.0);
+
+    // number of tests = 6
+}
+
+static void test_raw_soft_input(void){
+    double value = -255;
+
+    /* set soft channel */
+    testdbPutFieldOk("test_ai_rec.DTYP", DBF_STRING, "Raw Soft Channel");
+    testdbPutFieldOk("test_ai_rec.INP", DBF_STRING, "test_ai_link_rec.VAL");
+    testdbPutFieldOk("test_ai_link_rec.FLNK", DBF_STRING, "test_ai_rec");
+
+    /* set VAL to on linked record */
+    testdbPutFieldOk("test_ai_link_rec.VAL", DBF_DOUBLE, value);
+
+    /* verify that this record RVAL and VAL are updated */
+    testdbGetFieldEqual("test_ai_rec.VAL", DBF_DOUBLE, value);
+    testdbGetFieldEqual("test_ai_rec.RVAL", DBF_DOUBLE, value);
+
+    // number of tests = 6
+}
+
+static void test_operator_display(void){
+    const short hopr = 199;
+    const short lopr = 50;
+    const short prec = 2;
+    const char egu[] = "mm";
+    const char desc[] = "hmm?";
+
+    /* set operator display parameters */
+    testdbPutFieldOk("test_ai_rec.EGU", DBF_STRING, egu);
+    testdbPutFieldOk("test_ai_rec.DESC", DBF_STRING, desc);
+    testdbPutFieldOk("test_ai_rec.HOPR", DBF_SHORT, hopr);
+    testdbPutFieldOk("test_ai_rec.LOPR", DBF_SHORT, lopr);
+    testdbPutFieldOk("test_ai_rec.PREC", DBF_SHORT, prec);
+    
+    /* verify operator display parameters read back */
+    testdbGetFieldEqual("test_ai_rec.NAME", DBF_STRING, "test_ai_rec");
+    testdbGetFieldEqual("test_ai_rec.EGU", DBF_STRING, egu);   
+    testdbGetFieldEqual("test_ai_rec.DESC", DBF_STRING, desc);
+    testdbGetFieldEqual("test_ai_rec.HOPR", DBF_SHORT, hopr);
+    testdbGetFieldEqual("test_ai_rec.LOPR", DBF_SHORT, lopr);
+    testdbGetFieldEqual("test_ai_rec.PREC", DBF_SHORT, prec); 
+
+    // number of tests = 7
+}
+
+static void test_no_linr_unit_conversion(void){
+    const short roff = 10;
+    const short aslo = 2;
+    const short aoff = 4;
+
+    const short rval = 9;
+    const short val = (((rval + roff) * aslo) + aoff);
+
+
+    /* set soft channel */
+    testdbPutFieldOk("test_ai_rec.DTYP", DBF_STRING, "Raw Soft Channel");
+    testdbPutFieldOk("test_ai_rec.INP", DBF_STRING, "test_ai_link_rec.VAL");
+    testdbPutFieldOk("test_ai_link_rec.FLNK", DBF_STRING, "test_ai_rec");
+
+    /* set unit conversion parameters */
+    testdbPutFieldOk("test_ai_rec.ROFF", DBF_SHORT, roff);
+    testdbPutFieldOk("test_ai_rec.ASLO", DBF_SHORT, aslo);
+    testdbPutFieldOk("test_ai_rec.AOFF", DBF_SHORT, aoff);
```

Comment:
I know the default value of `LINR` is already `NO CONVERSION`, but it would be good to explicitly set it here, for the sake of clarity of the test routine.

Reply:
Ok. Correcting this.